player/screenshot.c

```diff
@@ -491,8 +492,11 @@ void cmd_screenshot(void *p)
 
     if (image) {
         char *filename = gen_fname(cmd, image_writer_file_ext(opts));
-        if (filename)
+        if (filename) {
             cmd->success = write_screenshot(cmd, image, filename, NULL);
+            node_init(res, MPV_FORMAT_NODE_MAP, NULL);
+            node_map_add_string(res, "filename", filename);
```

Comment:
Oh wait, I just thought of something. This should only be done if `cmd->success` is true (i.e. `write_screenshot` succeeds.

Reply:
ehhh, we'll just nest another if then.

Reply:
No problem with that imo there's a 3-level nested right above it.